### DISCUSSION TITLE
Run an existing Docker image, mount the local directory to it, and execute the command line options in /bin/bash

### DISCUSSION THREAD

**Discussion by kovar-ursa** [gabrieldemarmiesse/python-on-whales#602] [Q&A]
I've built a Docker image with various toolchains for compiling on different architectures. I used buildx to create a multi-architecture image and run this with:

`docker run -it -v $(pwd)/:/builddir docker.io/me/xcompile:latest@sha256:0a3d9cdcbc8c3364b884d697598e17437b360153d5d62f8fc55b62b19db8e777 make all`

So, mount the current directory into the image, run a specific version of the image, and execute `make all`.

What is the whales version of this where "make all" is a command line option?

I've tried this:

`    docker.run(
        ["docker.io/me/xcompile:latest@sha256:c8c5cc44c4a14c55d36e09cbdd3f175ca74a6b5ad5121a0f75346f66272228dd"],
        volumes=[pwd, "/builddir"],
        interactive=True,
        tty=True,
        detach=False
    )
`

The volumes section produces errors where both pwd and the "/builddir" look like: `/:b:u:i:l:d:d:i:r:`

This appears to be a LOT easier than using subprocess once I figure it out.

**Comment by gabrieldemarmiesse** [comment]
Hi there! Thanks for trying out this package! There is definitly improvements we could do in terms of checking the types of input, which is the error that you are hitting now.

I think you want something like this:
```python
 docker.run( "docker.io/me/xcompile:latest@sha256:c8c5cc44c4a14c55d36e09cbdd3f175ca74a6b5ad5121a0f75346f66272228dd", ["make", "all"], volumes=[(pwd, "/builddir")], interactive=True, tty=True, detach=False )
```

You can take a look at the docs for docker run here: https://gabrieldemarmiesse.github.io/python-on-whales/sub-commands/container/#python_on_whales.components.container.cli_wrapper.ContainerCLI.run 

let me know if you need more help!

**Reply by kovar-ursa** [reply]
That resulted in a similar error:

`python_on_whales.exceptions.DockerException: The docker command executed was `/usr/local/bin/docker container run --interactive --tty --volume ":/:U:s:e:r:s:/:me/:w:o:r:k:/:x:c:o:m:p:i:l:e:" --volume /:b:u:i:l:d:d:i:r `

Something is inserting colons in the string arguments to --volume.

**Reply by gabrieldemarmiesse** [reply]
Are you sure you copied the volume part of my command correctly? I changed it and made it into a tuple. That should have solved the issue

**Reply by kovar-ursa** [reply]
Closer! (And, many thanks for the help.)

`python_on_whales.exceptions.DockerException: The docker command executed was `/usr/local/bin/docker container run --interactive --tty --volume "/Users/me/work/xcompile":/builddir ['docker.io/me/xcompile:latest@sha256:c8c5cc44c4a14c55d36e09cbdd3f175ca74a6b5ad5121a0f75346f66272228dd'] ./configure; make clean; make all`.
It returned with code 125
The content of stdout can be found above the stacktrace (it wasn't captured).
The content of stderr can be found above the stacktrace (it wasn't captured).`

If I run the command directly it produces output that allows me to debug the issues. Using `whales` I only get the 125 error code.
